docs/src/pages/guides/interoperability/interoperability.md

```diff
@@ -599,3 +599,137 @@ It works exactly like styled components. You can [use the same guide](/guides/in
+
+3. Fix the CSS injection order
+
+**Note:** Most CSS-in-JS solutions inject their styles at the bottom of the HTML `<head>`, which gives MUI precedence over your custom styles. To remove the need for **!important**, you need to change the CSS injection order. Here's a demo of how it can be done in MUI:
+
+```jsx
+import * as React from 'react';
+import { StyledEngineProvider } from '@mui/material/styles';
+
+export default function GlobalCssPriority() {
+  return (
+    <StyledEngineProvider injectFirst>
+      {/* Your component tree. Now you can override MUI's styles. */}
+    </StyledEngineProvider>
+  );
+}
+```
+
+**Note:** If you are using emotion and have a custom cache in your app, that one will override the one coming from MUI. In order for the injection order to still be correct, you need to add the prepend option. Here is an example:
+
+```jsx
+import * as React from 'react';
+import { CacheProvider } from '@emotion/react';
+import createCache from '@emotion/cache';
+
+const cache = createCache({
+  key: 'css',
+  prepend: true,
+});
+
+export default function PlainCssPriority() {
+  return (
+    <CacheProvider value={cache}>
+      {/* Your component tree. Now you can override MUI's styles. */}
+    </CacheProvider>
+  );
+}
+```
+
+**Note:** If you are using styled-components and have `StyleSheetManager` with a custom `target`, make sure that the target is the first element in the HTML `<head>`. If you are curious to see how it can be done, you can take a look on the `StylesProvider` implementation in the `@mui/styled-engine-sc` package.
+
+Now you can start using Tailwind CSS on the MUI components
+
+**App.js**
+
+```jsx
+import * as React from 'react';
+import Slider from '@mui/material/Slider';
+
+export default function App() {
+  return (
+    <div>
+      <Slider defaultValue={30} className="color-indigo-500" />
+    </div>
+  );
+}
+```
+
+### Deeper elements
+
+If you attempt to style the Slider, you will likely need to affect some of the Slider's child elements, for example the thumb.
+
+The following examples override the slider's `thumb` style and adds custom styles for the active pseudo state.
+
+**SliderThumbOverrides.js**
+
+```jsx
+import * as React from 'react';
+import Slider from '@mui/material/Slider';
+
+export default function SliderThumbOverrides() {
+  return (
+    <Slider
+      className="p-4"
+      componentsProps={{ thumb: { className: 'hover:shadow-none' } }}
+    />
+  );
+}
+```
+
+### Styling pseudo states
+
+If you would like to style the component based on some pseudo state, you can use the appropariate key in the `classes` prop.
+Here is an example of how you can style the component, when the active state is on:
+
+**SliderPseudoStateOverrides.js**
+
+```jsx
+import * as React from 'react';
+import Slider from '@mui/material/Slider';
+
+export default function SliderThumbOverrides() {
+  return <Slider className="p-4" classes={{ active: 'shadow-none' }} />;
```

Comment:
This makes me think that `classes` prop is still useful. Joy might have to support classes as well?

Reply:
or if we want to drop `classes`, then I should be able to read the state from `componentProps`.

```js
<Slider
  className="p-4"
  componentsProps={{
    root: ({ ownerState }) => ({
      className: ownerState.active && 'shadow-none',
    })
  }}
/>
```

No need to discuss on this PR. We can revisit later.

Reply:
The important part here is how we want to support it in the base components, what would make most sense there. Your suggestion about making the `componentsProps` accept callback sounds great, more over because sometimes some props may depend on the state of the owner component. cc @michaldudak

Reply:
This is something I also realized recently - in Base (and Joy as well) we do need a way to customize class names applied automatically (or even disable them). The callback in componentsProps can work, but it's pretty verbose.
But in general yes, we can consider having a callback here. It may come in handy not just for settings className for other props as well.